### PR TITLE
[Packaging] Support Python 3.10

### LINKED ISSUE #19857: Support Python 3.10
## Description

Python 3.10 has been released on 2021-10-04: https://www.python.org/downloads/

Azure CLI needs to support Python 3.10 like how we supported 3.9 (https://github.com/Azure/azure-cli/issues/16611).

## TODOs

- [x] [Knack](https://github.com/microsoft/knack) supports Python 3.10: https://github.com/microsoft/knack/pull/250
- [x] [azdev](https://github.com/Azure/azure-cli-dev-tools) supports Python 3.10: https://github.com/Azure/azure-cli-dev-tools/pull/319
- [x] CI passes with Python 3.10
  - [x] Dependencies like `pytest`, `pylint` are updated to the Python 3.10 compatible version: https://github.com/Azure/azure-cli/pull/20192
  - [x] New `pylint` check failures are fixed or ignored: https://github.com/Azure/azure-cli/pull/20192
  - [x] All test cases pass: https://github.com/Azure/azure-cli/pull/20195

### PR DESCRIPTION
**Pull request by jiasli**
**Description**

Close #19857
Similar to #17368

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 1]
```diff
@@ -132,7 +132,7 @@ scp==0.13.2
 semver==2.13.0
 six==1.14.0
 sshtunnel==0.1.5
-tabulate==0.8.3
+tabulate==0.8.9
```
Comment (jiasli) [inline src/azure-cli/requirements.py3.Darwin.txt:135]: Old `tabulate` uses string to compare version, resulting in `"10" < "3"`, which is certainly wrong.

```
    if python_version_tuple() >= ("3", "3", "0"):
        from collections.abc import Iterable
    else:
>       from collections import Iterable
E       ImportError: cannot import name 'Iterable' from 'collections' (/opt/hostedtoolcache/Python/3.10.0/x64/lib/python3.10/collections/__init__.py)

env/lib/python3.10/site-packages/tabulate.py:16: ImportError
```

File: src/azure-cli-core/azure/cli/core/_help.py [hunk 2]
```diff
@@ -88,12 +88,9 @@ def _print_examples(help_file):
             print('')
 
     @staticmethod
-    def _print_az_find_message(command, enable_color):
-        from colorama import Style
+    def _print_az_find_message(command):
         indent = 0
         message = 'To search AI knowledge base for examples, use: az find "az {}"'.format(command)
-        if enable_color:
-            message = Style.BRIGHT + message + Style.RESET_ALL
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/_help.py:96]: No more color?

**Review by wangzelin007 — APPROVED** [review]
LGTM

File: src/azure-cli-core/azure/cli/core/_help.py (hunk 2 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_help.py:96]: The color makes this sentence too outstanding, outshining the original help message.

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/_help.py (hunk 2 above)
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_help.py:96]: We remove the color here as it outshines whole original help messages and examples.

File: src/azure-cli/setup.py [hunk 3]
```diff
@@ -136,6 +137,7 @@
     'azure-synapse-managedprivateendpoints~=0.3.0',
     'azure-synapse-spark~=0.2.0',
     'chardet~=3.0.4',
+    'colorama~=0.4.4',
```
Comment (jiasli) [inline src/azure-cli/setup.py:140]: `colorama` was removed from Knack for non-Windows systems (https://github.com/microsoft/knack/pull/249), but Azure CLI still needs it.

Comment (jiasli) [inline src/azure-cli/setup.py:140]: `colorama` was removed from Knack for non-Windows systems (https://github.com/microsoft/knack/pull/249), but `azure-cli` still needs it.

**Comment by jiasli** [comment]
Still seeing `DeprecationWarning`s from `pywin32`. Reported at https://github.com/mhammond/pywin32/issues/1802.

**Comment by jiasli** [comment]
Installing `psutil` fails without Microsoft Visual C++, and there is no fix yet.

- https://github.com/giampaolo/psutil/issues/2012
- https://github.com/giampaolo/psutil/issues/2012

**Pull request merged**